contracts/strategies/StrategyTUSDypool.sol

```diff
@@ -0,0 +1,365 @@
+    address public constant usdc = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant yusdc = address(0xd6aD7a6750A7593E092a9B218d66C0A814a3436e);
+    address public constant usdt = address(0xdAC17F958D2ee523a2206206994597C13D831ec7);
+    address public constant yusdt = address(0x83f798e925BcD4017Eb265844FDDAbb448f1707D);
+    address public constant tusd = address(0x0000000000085d4780B73119b644AE5ecd22b376);
+    address public constant ytusd = address(0x73a052500105205d34Daf004eAb301916DA8190f);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+    address public keeper;
+
+    uint256 public constant DENOMINATOR = 10000;
+    uint256 public treasuryFee = 500;
+    uint256 public withdrawalFee = 0;
+    uint256 public strategistReward = 50;
+    uint256 public threshold = 6000;
+    uint256 public slip = 100;
+    uint256 public tank = 0;
+    uint256 public p = 0;
+
+    event Threshold(address indexed strategy);
+
+    modifier isAuthorized() {
+        require(
+            msg.sender == governance ||
+                msg.sender == strategist ||
+                msg.sender == keeper ||
+                msg.sender == controller ||
+                msg.sender == address(this),
+            "!authorized"
+        );
+        _;
+    }
+
+    constructor(address _controller) public {
+        governance = msg.sender;
+        strategist = msg.sender;
+        keeper = msg.sender;
+        controller = _controller;
+    }
+
+    function getName() external pure returns (string memory) {
+        return "StrategyTUSDypool";
+    }
+
+    function deposit() public isAuthorized {
+        rebalance();
+        uint256 _want = (IERC20(want).balanceOf(address(this))).sub(tank);
+        if (_want > 0) {
+            IERC20(want).safeApprove(ytusd, 0);
+            IERC20(want).safeApprove(ytusd, _want);
+            yvERC20(ytusd).deposit(_want);
+        }
+        uint256 _y = IERC20(ytusd).balanceOf(address(this));
+        if (_y > 0) {
+            IERC20(ytusd).safeApprove(ypool, 0);
+            IERC20(ytusd).safeApprove(ypool, _y);
+            uint256 v = _want.mul(1e18).div(ICurveFi(ypool).get_virtual_price());
+            ICurveFi(ypool).add_liquidity([0, 0, 0, _y], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        }
+        uint256 _bal = IERC20(ycrv).balanceOf(address(this));
+        if (_bal > 0) {
+            IERC20(ycrv).safeApprove(yycrv, 0);
+            IERC20(ycrv).safeApprove(yycrv, _bal);
+            yvERC20(yycrv).deposit(_bal);
+        }
+    }
+
+    // Controller only function for creating additional rewards from dust
+    function withdraw(IERC20 _asset) external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        require(want != address(_asset), "want");
+        require(ycrv != address(_asset), "ycrv");
```

Comment:
Previous version had this line also
`require(y != address(_asset), "y");` where "y" is the TUSD y vault here 0x73a052500105205d34Daf004eAb301916DA8190f

https://github.com/iearn-finance/yearn-protocol/blob/b21d9613e74998455ddf81d829547cf491621f9d/contracts/strategies/StrategyTUSDCurve.sol#L77

is this ok to be removed?

Reply:
It is always consumed after auto-balancing, so it can be removed, maintaining less overhead. And in an emergent case, this case is also covered in conversion.